lua/orgmode/utils/init.lua

```diff
@@ -576,6 +577,7 @@ function utils.detect_filetype(name, skip_ftmatch)
     shell = 'bash',
     uxn = 'uxntal',
   }
+  map = vim.tbl_deep_extend('force', map, config.org_edit_src_filetype_map)
```

Comment:
Let's move this near the end of the function before the last `return`. Also, please switch up a logic like this:

```lua
  if config.org_edit_src_filetype_map[name] then
    return config.org_edit_src_filetype_map[name]
  end

  return name:lower()
```

This function is called a lot through a custom directive when rendering blocks, so i'd rather keep it as optimized as possible.